qmcpy/discrete_distribution/digital_net_b2/digital_net_b2.py

```diff
@@ -289,26 +321,45 @@ def gen_samples(self, n=None, n_min=0, n_max=8, warn=True, return_unrandomized=F
         if n:
             n_min = 0
             n_max = n
-        if n_min == 0 and self.set_rshift==False and warn:
+        if n_min == 0 and self.set_rshift==False and warn and self.set_owen == False:
             warnings.warn("Non-randomized DigitalNetB2 sequence includes the origin",ParameterWarning)
         if return_unrandomized and not self.set_rshift:
             raise ParameterError("return_unrandomized=True only applies when randomize includes a digital shift.")
         n = int(n_max-n_min)
+        xb = zeros((n,self.d),dtype=uint64)
         x = zeros((n,self.d),dtype=double)
         xr = zeros((n,self.d),dtype=double)
-        rc = self.dnb2_cf(int(n_min),n,self.d,self.graycode,self.m_max,self.t_lms,self.z,self.set_rshift,self.rshift,x,xr)
+        rc = self.dnb2_cf(int(n_min),n,self.d,self.graycode,self.m_max,self.t_lms,self.z,self.set_rshift,self.rshift,xb,x,xr)
         if rc!=0:
             raise ParameterError(self.errors[rc])
         if return_unrandomized:
             return xr,x
         elif self.set_rshift:
             return xr
+        elif self.set_owen:
+            return self.owen_scr(xb)
         else:
             return x
     
     def pdf(self, x):
         """ pdf of a standard uniform """
-        return ones(x.shape[0], dtype=float)        
+        return ones(x.shape[0], dtype=float) 
+
+    def owen_scr(self,xbs): 
+        """ generate samples based on Nested Uniform Scrambling (Owen Scrambling)"""
+        n = xbs.shape[0]
+        xbrs_fin = zeros((n,self.d),dtype=uint64)
+        for j in range(self.d):
+            xbrs = zeros(n)
```

Comment:
assign directly to `xbrs_fin` rather than assigning `xb` to `xbs` then `xbs` to `xbrs_fin`